src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -36,12 +36,51 @@ import tooltip from 'ui/tooltip';
  * Hide the tooltip for the text input
  * @param {jQuery} $input
  */
-var hideTooltip = function hideTooltip($input) {
+const hideTooltip = function hideTooltip($input) {
     if ($input.data('$tooltip')) {
         $input.data('$tooltip').hide();
     }
 };
 
+const DELIMETER_SYMBOL = '.';
+const MINUS_SYMBOL = '-';
+const cleanupIntegerRegExp = new RegExp('\\D', 'g') // Remove all non-digits except the delimeter
```

Comment:
What about e notation? like `5000` === `5e3` both are valid integers. And there is possibility to setup base, that could be 16 so `0x1A` would be also valid input for int